File: app/src/main/java/com/example/reminders20/db/ReminderDao.java
package com.example.reminders20.db;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;

import java.util.List;

import io.reactivex.rxjava3.core.Completable;

@Dao
public interface ReminderDao{
    @Query("SELECT * FROM Reminder")
    LiveData<List<Reminder>> getAll();
    @Query("SELECT * FROM Reminder WHERE timestamp = :timestamp")
    LiveData<Reminder> getReminderByTimestamp(long timestamp);
    @Insert (onConflict = OnConflictStrategy.REPLACE)
    Completable insertReminder(Reminder reminder);
    @Delete
    Completable deleteReminder(Reminder reminder);
}

File: app/src/main/java/com/example/reminders20/db/ReminderDao.kt
package com.example.reminders20.db

import androidx.lifecycle.LiveData
import androidx.room.*
import kotlinx.coroutines.flow.Flow

@Dao
interface ReminderDao {
    @Query("SELECT * FROM Reminder ORDER BY timestamp DESC")
    fun getAll(): Flow<List<Reminder>>

    @Query("SELECT * FROM Reminder WHERE timestamp = :timestamp")
    fun getReminderByTimestamp(timestamp: Long): LiveData<Reminder?>

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun insertReminder(reminder: Reminder)

    @Delete
    suspend fun deleteReminder(reminder: Reminder)
}